pkg/buildpod/generator_test.go

```diff
@@ -220,6 +277,155 @@ func testGenerator(t *testing.T, when spec.G, it spec.S) {
+					`{ "stack": { "runImage": { "image": "some-registry.io/run-image"} } }`)
+				require.NoError(t, err)
+
+				image, err = imagehelpers.SetStringLabel(image, cnb.BuilderMetadataLabel, //language=json
+					`{
+			 "stack": {
+			   "runImage": {
+			     "image": "some-registry.io/run-image"
+			   }
+			 },
+			 "lifecycle": {
+			   "version": "0.9.0",
+			   "api": {
+			     "buildpack": "0.7",
+			     "platform": "0.5"
+			   }
+			 }
+			}`)
+				require.NoError(t, err)
+
+				image, err = imagehelpers.SetEnv(image, "CNB_USER_ID", "1234")
+				require.NoError(t, err)
+
+				image, err = imagehelpers.SetEnv(image, "CNB_GROUP_ID", "5678")
+				require.NoError(t, err)
+
+				imageFetcher.AddImage("some/builde@sha256:1b2911dd8eabb4bdb0bda6705158daa4149adb5ca59dc990146772c4c6deecb4", image, keychain)
+
+				buildPodConfig = v1alpha1.BuildPodImages{}
+				generator = &buildpod.Generator{
+					BuildPodConfig:  buildPodConfig,
+					K8sClient:       fakeK8sClient,
+					KeychainFactory: keychainFactory,
+					ImageFetcher:    imageFetcher,
+				}
+
+				build = &testBuildPodable{
+					serviceAccount: serviceAccountName,
+					namespace:      namespace,
+					buildBuilderSpec: v1alpha1.BuildBuilderSpec{
+						Image:            "some/builde@sha256:1b2911dd8eabb4bdb0bda6705158daa4149adb5ca59dc990146772c4c6deecb4",
+						ImagePullSecrets: builderPullSecrets,
+					},
+				}
+			})
+			it("returns a list of taints present on all windows nodes", func() {
+				pod, err := generator.Generate(build)
+				require.NoError(t, err)
+				assert.NotNil(t, pod)
+
+				assert.Equal(t, []buildPodCall{{
+					BuildPodImages: buildPodConfig,
+					Secrets: []corev1.Secret{
+						*gitSecret,
+						*dockerSecret,
+					},
+					BuildPodBuilderConfig: v1alpha1.BuildPodBuilderConfig{
+						StackID:     "some.stack.id",
+						RunImage:    "some-registry.io/run-image",
+						Uid:         1234,
+						Gid:         5678,
+						PlatformAPI: "0.5",
+						OS:          "windows",
+						NodeTaints: []v1.Taint{
+							{
+								Key:    "test-key",
+								Value:  "test-value",
+								Effect: corev1.TaintEffectNoSchedule,
+							},
+						},
+					},
+				}}, build.buildPodCalls)
+			})
+			it("returns an empty list any taints are different across windows nodes", func() {
+				windowsNode3 := &corev1.Node{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "windows-node-3",
+						Labels: map[string]string{
+							"kubernetes.io/os": "windows",
+						},
+					},
+					Spec: corev1.NodeSpec{
+						Taints: []corev1.Taint{
+							{
+								Key:       "different-key",
+								Value:     "different-value",
+								Effect:    corev1.TaintEffectNoSchedule,
+								TimeAdded: nil,
+							},
+						},
+					},
+				}
+
+				fakeK8sClientWithDifferentNode := fake.NewSimpleClientset(serviceAccount, dockerSecret, gitSecret, ignoredSecret, linuxNode, windowsNode1, windowsNode2, windowsNode3)
+				generator.K8sClient = fakeK8sClientWithDifferentNode
```

Comment:
You could also `fakeK8sClient.CoreV1().Nodes().Create()` to add to the existing client.

Reply:
oh sweet thanks